src/app/examples/form/components/form.component.ts

```diff
@@ -0,0 +1,126 @@
+import { Component, OnInit, ViewChild, OnDestroy } from '@angular/core';
+import {
+  Validators,
+  FormBuilder,
+  AbstractControl,
+  FormGroup,
+  FormControl
+} from '@angular/forms';
+import { CdkTextareaAutosize } from '@angular/cdk/text-field';
+import { ROUTE_ANIMATIONS_ELEMENTS } from '@app/core';
+import { Store, select } from '@ngrx/store';
+import { Subject } from 'rxjs';
+import { State } from '@app/examples/examples.state';
+import { takeUntil } from 'rxjs/operators';
+import { selectForm } from '@app/examples/form/form.selectors';
+import { ActionFormRetrieve } from '@app/examples/form/form.actions';
+import { Form } from '@app/examples/form/form.model';
+import { MatSnackBar } from '@angular/material';
+
+@Component({
+  selector: 'anms-form',
+  templateUrl: './form.component.html',
+  styleUrls: ['./form.component.scss']
+})
+export class FormComponent implements OnInit, OnDestroy {
+  private unsubscribe$: Subject<void> = new Subject<void>();
+  initialized;
+
+  @ViewChild('autosize') autosize: CdkTextareaAutosize;
+
+  routeAnimationsElements = ROUTE_ANIMATIONS_ELEMENTS;
+
+  form = this.fb.group({
+    autosave: true,
+    firstname: ['', [Validators.required]],
+    lastname: ['', [Validators.required]],
+    email: ['', [Validators.required, Validators.email]],
+    description: [
+      '',
+      [
+        Validators.required,
+        Validators.minLength(10),
+        Validators.maxLength(1000)
+      ]
+    ],
+    wannagift: [''],
```

Comment:
Can be please changed to something more formal like `requestGift` or `acceptsGift`, ...